### PR TITLE
Make_unique Homework

### PR DESCRIPTION
**Pull request by mateusz-talma**

### CODE REVIEW

**Review by majkel84 — APPROVED** [review]

**Review by MaciejSalamonski — APPROVED** [review]
Good job :+1:

**Review by WBigaj — APPROVED** [review]
I have checked your code and it looks good 👍

File: module1/homework/make_unique/makeUnique-ut.cpp [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+TEST(makeUnique, shouldCreatePointerToIntWithDefaultValue) {
+    auto ptr = myMakeUnique::make_unique<int>();
+
+    ASSERT_EQ(*ptr, int());
+}
+
+TEST(makeUnique, shouldCreatePointerToTuple) {
+    auto ptr = myMakeUnique::make_unique<std::tuple<int, double>>(
+        std::make_tuple(TestConsts::intValue, TestConsts::doubleValue));
```
Comment (ziobron) [inline module1/homework/make_unique/makeUnique-ut.cpp]: std::tuple with std::make_tuple is redundant. std::make_tuple returns just std::tuple. Get rid of one of them.

File: module1/homework/make_unique/makeUnique-ut.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    ASSERT_EQ(*ptr, std::make_tuple(TestConsts::intValue, TestConsts::doubleValue));
+}
+
+TEST(makeUnique, shouldCreatePointerToOwnStructure) {
+    auto ptr = myMakeUnique::make_unique<TestData>(TestData(TestConsts::intValue,
+                                                            TestConsts::doubleValue,
+                                                            TestConsts::boolValue));
```
Comment (ziobron) [inline module1/homework/make_unique/makeUnique-ut.cpp]: It's still only one argument that is passed to make_unique and it is TestData. Please pass 3 arguments and create (not copy or move) TestData inside make_unique. So you have only cases when there is either 1 or 0 parameters passed. I need a proof in tests that your function can take any number of arguments :)
```suggestion
    auto ptr = myMakeUnique::make_unique<TestData>(TestConsts::intValue,
                                                   TestConsts::doubleValue,
                                                   TestConsts::boolValue);
```

**Review by ziobron — CHANGES_REQUESTED** [review]

**Review by ziobron — APPROVED** [review]

**Comment by ziobron** [comment]
{ "requirement_ids": [1119, 4443], "points": [3, 40] }

**Pull request closed without merging**